cmd/sonobuoy/app/runner.go

```diff
@@ -0,0 +1,330 @@
+package app
+
+import (
+	"archive/tar"
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"path"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"golang.org/x/sync/errgroup"
+	"k8s.io/client-go/util/exec"
+
+	"github.com/heptio/sonobuoy/pkg/client"
+	"github.com/heptio/sonobuoy/pkg/errlog"
+	"github.com/heptio/sonobuoy/pkg/plugin/aggregation"
+)
+
+type runnerFlags struct {
+	genFlags
+	showAll         bool
+	skipPreflight   bool
+	pollingInterval time.Duration
+	statusLoopDelay time.Duration
+	retrievalDelay  time.Duration
+	timeout         time.Duration
+}
+
+var (
+	runnerflags    runnerFlags
+	sonobuoyClient *client.SonobuoyClient
+)
+
+func RunnerFlagSet(cfg *runnerFlags) *pflag.FlagSet {
+	runnerset := pflag.NewFlagSet("runner", pflag.ExitOnError)
+	// Default to detect since we need kubeconfig regardless
+	runnerset.AddFlagSet(GenFlagSet(&cfg.genFlags, DetectRBACMode, ConformanceImageVersionAuto))
+	AddSkipPreflightFlag(&cfg.skipPreflight, runnerset)
+
+	runnerset.DurationVar(
+		&cfg.timeout, "runner-timeout", 6*time.Hour,
+		"Length of time to give the runner before giving up. (Default: 6h)",
```

Comment:
where did these times come from?

Reply:
Just some arbitrary time for how long some of our Sonobuoy runs have taken. If you have a better default timeout I'm definitely open for suggestions.